pyanaconda/packaging/__init__.py

```diff
@@ -290,7 +290,11 @@ def selectGroup(self, groupid, default=True, optional=False):
         if grp in self.data.packages.excludedGroupList:
             self.data.packages.excludedGroupList.remove(grp)
 
-        self.data.packages.groupList.append(grp)
+        # If the group is part of the environment, no need to duplicate it in
+        # the group list
+        if self.data.packages.environment and \
+                groupid not in self.environmentGroups(self.data.packages.environment):
+            self.data.packages.groupList.append(grp)
```

Comment:
Is it okay that we don't add the group to the list if no environment is selected? I'd expect it to be added so please extend the comment a bit to explain why that's not happening.